src/main/java/org/terasology/tutorialWorldGenerationLakes/LakesRasterizer.java

```diff
@@ -21,13 +21,16 @@
 import org.terasology.world.block.Block;
 import org.terasology.world.block.BlockManager;
 import org.terasology.world.chunks.CoreChunk;
+import org.terasology.world.generation.Facet;
 import org.terasology.world.generation.Region;
+import org.terasology.world.generation.Requires;
 import org.terasology.world.generation.WorldRasterizerPlugin;
 import org.terasology.world.generation.facets.SeaLevelFacet;
 import org.terasology.world.generation.facets.SurfaceHeightFacet;
 import org.terasology.world.generator.plugin.RegisterPlugin;
 
 @RegisterPlugin
+@Requires({@Facet(SeaLevelFacet.class), @Facet(SurfaceHeightFacet.class)})
```

Comment:
Curiously, is this change related, or just happened to stick out as needed?

Reply:
This should not change anything as of now, however in theory this should be required on rasterizers/plugins that utilize other facets and do not have them straight away in `@Updates`, as I believe that there is no guarantee in engine API that all the facetProviders will be run before rasterizers, or that they will be run at all even if not explicitly needed (which would for sure provide room for optimization, if only facetProviders for required facets were run)